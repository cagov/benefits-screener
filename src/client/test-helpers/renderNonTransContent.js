import React from "react";
import { shallow } from "enzyme";
import { I18nextProvider, Trans } from "react-i18next";
import i18n from "./i18nForTests";
import shallowUntilTarget from "./shallowUntilTarget";

function renderNonTransContent(Component, props = {}) {
  const providersWrapper = shallow(
    <I18nextProvider i18n={i18n}>
      <Component {...props} />
    </I18nextProvider>
  );

  const wrapper = shallowUntilTarget(providersWrapper, Component.name);
  // Incorrectly using renderNonTransContent to render a component that
  // contains a Trans component results in a silent failure (the rendered
  // snapshot will contain the Trans component itself instead of the rendered
  // contents). This test makes that silent failure more obvious.
  expect(wrapper.containsMatchingElement(<Trans />)).toBe(false);

  return wrapper;
}

module.exports = renderNonTransContent;
