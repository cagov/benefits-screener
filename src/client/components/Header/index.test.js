import renderNonTransContent from "../../test-helpers/renderNonTransContent";
import Component from "./index";

describe("<Header />", () => {
  it("renders the component", async () => {
    const wrapper = renderNonTransContent(Component, "Header");

    expect(wrapper).toMatchSnapshot();
  });
});
