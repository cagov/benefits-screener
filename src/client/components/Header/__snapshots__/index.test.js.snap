// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders the header component 1`] = `
<header
  className="header border-bottom border-secondary"
>
  <Navbar
    bg="primary"
    className="justify-content-between"
    collapseOnSelect={false}
    expand={true}
    variant="custom"
  >
    <NavbarBrand
      href="https://ca.gov"
    >
      <img
        alt="California gov logo"
        className="d-inline-block align-top"
        height="30"
        src="images/Ca-Gov-Logo-Gold.svg"
        width="30"
      />
    </NavbarBrand>
    <Nav
      fill={false}
      justify={false}
    >
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        href="https://edd.ca.gov"
      >
        <img
          alt="home icon"
          height="15"
          src="images/home.svg"
          width="15"
        />
         
        <span>
          Home
        </span>
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        href="https://edd.ca.gov/login.htm"
      >
        <img
          alt="key icon"
          height="15"
          src="images/key.svg"
          width="15"
        />
         
        <span>
          Log In
        </span>
      </NavLink>
    </Nav>
  </Navbar>
  <Navbar
    className=""
    collapseOnSelect={true}
    expand="md"
    variant="light"
  >
    <NavbarBrand
      href="https://edd.ca.gov"
    >
      <img
        alt="California gov logo"
        className="d-inline-block align-top"
        height="50"
        src="images/edd-logo-2-Color.svg"
        width="150"
      />
    </NavbarBrand>
    <NavbarToggle
      aria-controls="responsive-navbar-nav"
      label="Toggle navigation"
    />
    <NavbarCollapse
      className="justify-content-between"
    >
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        href="https://edd.ca.gov/jobs.htm"
      >
        <span
          className="text-dark"
        >
          Jobs
        </span>
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        href="https://edd.ca.gov/claims.htm"
      >
        <span
          className="text-dark"
        >
          Claims
        </span>
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        href="https://edd.ca.gov/employers.htm"
      >
        <span
          className="text-dark"
        >
          Employers
        </span>
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        href="https://edd.ca.gov/newsroom.htm"
      >
        <span
          className="text-dark"
        >
          Newsroom
        </span>
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        href="https://edd.ca.gov/serp.html?q="
      >
        <span
          className="text-dark"
        >
          Search
        </span>
      </NavLink>
    </NavbarCollapse>
  </Navbar>
</header>
`;
